solutions/LC Find Pivot Index.mdx

```diff
@@ -0,0 +1,192 @@
+---
+id: lc-find-pivot-index
+source: LeetCode
+title: Find Pivot Index
+author: Qi Wang
+---
+
+**Time Complexity:** $\mathcal O(N)$.
+
+## Solution 1
+
+The pivot index must have the same sum on its left and its right. To calculate this, we can create a prefix sum of the $\texttt{nums}$ array (in my solution it is cushioned with one $0$ at index $0$). Afterwards, all we need to do is to loop through every index from $0$ (inclusive) to $\texttt{nums}.length$ (exclusive) in the $\texttt{psums}$ array and check for:
+
+$$
+\texttt{psums}[i] - \texttt{psums}[0] == \texttt{psums}[\texttt{nums}.length] - \texttt{psums}[i+1]
+$$
+
+If this is true, we can return our current index because the problem wants the left-most index. If it finishes the for-loop without returning, it means there is no valid pivot index so return $-1$.
+
+## Solution 2
+
+The alternate solution does not require an array to store the sums. Instead, we only need the total sum of the array and the current sum at each index (exclusive of index). With this we check for:
+
+$$
+\texttt{curSum} == \texttt{total} - \texttt{curSum} - \texttt{nums}[i];
```

Comment:
>  - \texttt{nums}[i]

just leave it as curSum == total - curSum to make it less confusing

Reply:
also fix the respective code.

Reply:
@andrewwangva but for [1,7,3,6,5,6], the answer is at index 3. the left sum nor right sum include index 3 so how is curSum == total - curSum gonna work??? am i missing smth?